### PR TITLE
Set up CI with Azure Pipelines

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by mcfadden8** [comment]
@davidbeckingsale, LLNL.Umpire Failing?

**Comment by davidbeckingsale** [comment]
Whoops,  I messed up merging `develop`...

**Comment by davidbeckingsale** [comment]
@mcfadden8 @jdahm - this is ready to go!

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -38,11 +38,13 @@ if (ENABLE_C)
...
-  SOURCES vector_allocator.cpp
-  DEPENDS_ON umpire
-  OUTPUT_DIR ${UMPIRE_EXAMPLES_OUTPUT_DIR})
+if (NOT MSVC)
+  blt_add_executable(
+    NAME vector_allocator
+    SOURCES vector_allocator.cpp
+    DEPENDS_ON umpire
+    OUTPUT_DIR ${UMPIRE_EXAMPLES_OUTPUT_DIR})
+endif()
```
Comment (mcfadden8) [inline examples/CMakeLists.txt]: Please add comment for why this does not work on Windows.

File: examples/tutorial/CMakeLists.txt [hunk 2]
```diff
@@ -62,11 +62,13 @@ blt_add_executable(
...
-  SOURCES tut_typed_allocator.cpp
-  DEPENDS_ON umpire
-  OUTPUT_DIR ${tutorial_dir})
+if (NOT MSVC)
+  blt_add_executable(
+    NAME tut_typed_allocator
+    SOURCES tut_typed_allocator.cpp
+    DEPENDS_ON umpire
+    OUTPUT_DIR ${tutorial_dir})
+endif ()
```
Comment (mcfadden8) [inline examples/tutorial/CMakeLists.txt]: Please add comment for why this does not work on Windows.

File: src/umpire/ResourceManager.inl [hunk 3]
```diff
@@ -40,12 +44,21 @@ Allocator ResourceManager::makeAllocator(
 
     UMPIRE_LOG(Debug, "(name=\"" << name << "\")");
 
+#if defined(_MSC_VER)
+    UMPIRE_REPLAY("makeAllocator_attempt,"
+        << typeid(Strategy).name()
```
Comment (mcfadden8) [inline src/umpire/ResourceManager.inl]: Nice!

Does replay work with this?

**Comment by davidbeckingsale** [comment]
@mcfadden8 - TypedAllocator is working 😄

File: src/umpire/TypedAllocator.inl [hunk 4]
```diff
@@ -25,6 +25,13 @@ TypedAllocator<T>::TypedAllocator(Allocator allocator) :
 {
 }
 
+template<typename T>
+template<typename U>
+TypedAllocator<T>::TypedAllocator(const TypedAllocator<U>& other) :
+  m_allocator(other.m_allocator)
```
Comment (jdahm) [inline src/umpire/TypedAllocator.inl]: Do we use this somewhere?

Comment (davidbeckingsale) [reply src/umpire/TypedAllocator.inl]: It's used when creating a vector with a TypedAllocator with MSVC. Don't ask me why...

**Pull request merged**